### PR TITLE
Remove API Key mentions + deprioritize content

### LINKED ISSUE #3002: Update `Authenticate an automation tool`

### LINKED ISSUE #3003: Sidebar navigation changes
Pending decision made by docs team about where to move https://docs.astronomer.io/astro/api-keys#docusaurus_skipToContent_fallback

- [ ] Option 1: Remove doc entirely from sidebar navigation. Accessible by link only. 
- [ ] Option 2: Change title to include `(Deprecated)` and move to `References`
- [ ] Option 3: Change title to include `(Deprecated)` and move to a child of `Authenticate an automation tool to Astro`
Screenshot 2023-10-24 at 11 45 45 AM

### LINKED ISSUE #3004: Find + Replace all API Key mentions
Follow instructions on identified pages in [API Key Deprecation triage](https://docs.google.com/spreadsheets/d/10RrA3elnXx4HHEHFcFcDMV85awPdoRfi-XurPymwguc/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by lzdanski**
resolves #3004 
resolves #3003 
resolves #3002 

- Moves API Keys page to de-prioritized location in sidebar navigation
- Restructures/rewrites section about API keys and tokens recommended pathways (https://docs.astronomer.io/astro/airflow-api#step-1-retrieve-your-access-token)
- Removes API Key mentions for deprecation

### CODE REVIEW

File: sidebarsAstro.js [hunk 1]
```diff
@@ -122,7 +121,12 @@ module.exports = {
       label: "Automation & CI/CD",
       items: [
         "automation-overview",
-        "automation-authentication",
+        {
+        type: "category",
+        label: "Secrets backend",
```
Comment (jwitz) [inline sidebarsAstro.js:126]: I assume this was a copy paste error, but I also don't know what I'd call this category. I'd prefer to just put API keys at the very bottom of the "Automation & CI/CD" menu without a submenu, but also open to another title for this submenu.

**Review by jwitz — CHANGES_REQUESTED** [review]
One tiny comment about the sidebar, otherwise LGTM!

File: astro/deployment-settings.md [hunk 2]
```diff
@@ -247,5 +247,5 @@ Every Deployment has two PgBouncer Pods assigned to two different nodes to preve
 ## See also
 
 - [Set environment variables on Astro](environment-variables.md).
```
Comment (lzdanski) [inline astro/deployment-settings.md:249]:
```suggestion
- [Set environment variables on Astro](environment-variables.md)
```

**Review by jwitz — APPROVED** [review]

File: astro/cli/astro-deploy.md
Comment (lzdanski) [inline astro/cli/astro-deploy.md:1]: @sunkickr  - I think this is the only CLI related one that really needs a 👍 / 👎  for the adjustment from API Keys to API Tokens.

**Pull request merged**